// angular-bootstrap-md/esm2015/lib/free/utilities/facade/browser.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @license
 * Copyright Google Inc. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */
/**
 * JS version of browser APIs. This library can only run in the browser.
 * @type {?}
 */
const win = (typeof window !== 'undefined' && window) || (/** @type {?} */ ({}));
export { win as window };
/** @type {?} */
export const document = win.document;
/** @type {?} */
export const location = win.location;
/** @type {?} */
export const gc = win.gc ? (/**
 * @return {?}
 */
() => win.gc()) : (/**
 * @return {?}
 */
() => null);
/** @type {?} */
export const performance = win.performance ? win.performance : null;
/** @type {?} */
export const Event = win.Event;
/** @type {?} */
export const MouseEvent = win.MouseEvent;
/** @type {?} */
export const KeyboardEvent = win.KeyboardEvent;
/** @type {?} */
export const EventTarget = win.EventTarget;
/** @type {?} */
export const History = win.History;
/** @type {?} */
export const Location = win.Location;
/** @type {?} */
export const EventListener = win.EventListener;
