// @kamiazya/ngx-speech-recognition/esm2015/lib/service/speech-recognition.token.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InjectionToken } from '@angular/core';
/** @type {?} */
export const SpeechRecognitionGrammars = new InjectionToken('speech-recognition.grammars');
/** @type {?} */
export const SpeechRecognitionLang = new InjectionToken('speech-recognition.lang');
/** @type {?} */
export const SpeechRecognitionContinuous = new InjectionToken('speech-recognition.continuous');
/** @type {?} */
export const SpeechRecognitionInterimResults = new InjectionToken('speech-recognition.interimResults');
/** @type {?} */
export const SpeechRecognitionMaxAlternatives = new InjectionToken('speech-recognition.maxAlternatives');
/** @type {?} */
export const SpeechRecognitionServiceUri = new InjectionToken('speech-recognition.serviceURI');
/** @type {?} */
export const SpeechRecognitionAudiostartHandler = new InjectionToken('speech-recognition.onaudiostart');
/** @type {?} */
export const SpeechRecognitionSoundstartHandler = new InjectionToken('speech-recognition.onsoundstart');
/** @type {?} */
export const SpeechRecognitionSpeechstartHandler = new InjectionToken('speech-recognition.onspeechstart');
/** @type {?} */
export const SpeechRecognitionSpeechendHandler = new InjectionToken('speech-recognition.onspeechend');
/** @type {?} */
export const SpeechRecognitionSoundendHandler = new InjectionToken('speech-recognition.onsoundend');
/** @type {?} */
export const SpeechRecognitionAudioendHandler = new InjectionToken('speech-recognition.onaudioend');
/** @type {?} */
export const SpeechRecognitionResultHandler = new InjectionToken('speech-recognition.onresult');
/** @type {?} */
export const SpeechRecognitionNomatchHandler = new InjectionToken('speech-recognition.onnomatch');
/** @type {?} */
export const SpeechRecognitionErrorHandler = new InjectionToken('speech-recognition.onerror');
/** @type {?} */
export const SpeechRecognitionStartHandler = new InjectionToken('speech-recognition.onstart');
/** @type {?} */
export const SpeechRecognitionEndHandler = new InjectionToken('speech-recognition.onend');
