// angular-bootstrap-md/esm5/lib/free/utils/facade/browser.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*tslint:disable */
/**
 * @license
 * Copyright Google Inc. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */
/**
 * JS version of browser APIs. This library can only run in the browser.
 * @type {?}
 */
var win = typeof window !== 'undefined' && window || (/** @type {?} */ ({}));
export { win as window };
/** @type {?} */
export var document = win.document;
/** @type {?} */
export var location = win.location;
/** @type {?} */
export var gc = win['gc'] ? (/**
 * @return {?}
 */
function () { return win['gc'](); }) : (/**
 * @return {?}
 */
function () { return null; });
/** @type {?} */
export var performance = win['performance'] ? win['performance'] : null;
/** @type {?} */
export var Event = win['Event'];
/** @type {?} */
export var MouseEvent = win['MouseEvent'];
/** @type {?} */
export var KeyboardEvent = win['KeyboardEvent'];
/** @type {?} */
export var EventTarget = win['EventTarget'];
/** @type {?} */
export var History = win['History'];
/** @type {?} */
export var Location = win['Location'];
/** @type {?} */
export var EventListener = win['EventListener'];
/** @type {?} */
export var navigator = win['navigator'];
