// @kamiazya/ngx-speech-recognition/esm5/lib/service/speech-recognition.token.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InjectionToken } from '@angular/core';
/** @type {?} */
export var SpeechRecognitionGrammars = new InjectionToken('speech-recognition.grammars');
/** @type {?} */
export var SpeechRecognitionLang = new InjectionToken('speech-recognition.lang');
/** @type {?} */
export var SpeechRecognitionContinuous = new InjectionToken('speech-recognition.continuous');
/** @type {?} */
export var SpeechRecognitionInterimResults = new InjectionToken('speech-recognition.interimResults');
/** @type {?} */
export var SpeechRecognitionMaxAlternatives = new InjectionToken('speech-recognition.maxAlternatives');
/** @type {?} */
export var SpeechRecognitionServiceUri = new InjectionToken('speech-recognition.serviceURI');
/** @type {?} */
export var SpeechRecognitionAudiostartHandler = new InjectionToken('speech-recognition.onaudiostart');
/** @type {?} */
export var SpeechRecognitionSoundstartHandler = new InjectionToken('speech-recognition.onsoundstart');
/** @type {?} */
export var SpeechRecognitionSpeechstartHandler = new InjectionToken('speech-recognition.onspeechstart');
/** @type {?} */
export var SpeechRecognitionSpeechendHandler = new InjectionToken('speech-recognition.onspeechend');
/** @type {?} */
export var SpeechRecognitionSoundendHandler = new InjectionToken('speech-recognition.onsoundend');
/** @type {?} */
export var SpeechRecognitionAudioendHandler = new InjectionToken('speech-recognition.onaudioend');
/** @type {?} */
export var SpeechRecognitionResultHandler = new InjectionToken('speech-recognition.onresult');
/** @type {?} */
export var SpeechRecognitionNomatchHandler = new InjectionToken('speech-recognition.onnomatch');
/** @type {?} */
export var SpeechRecognitionErrorHandler = new InjectionToken('speech-recognition.onerror');
/** @type {?} */
export var SpeechRecognitionStartHandler = new InjectionToken('speech-recognition.onstart');
/** @type {?} */
export var SpeechRecognitionEndHandler = new InjectionToken('speech-recognition.onend');
