// @kamiazya/ngx-speech-recognition/esm5/public_api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*
 * Public API Surface of ngx-speech-recognition
 */
import './lib/polifill';
export {} from './lib/adapter';
export { SpeechRecognitionGrammars, SpeechRecognitionLang, SpeechRecognitionContinuous, SpeechRecognitionInterimResults, SpeechRecognitionMaxAlternatives, SpeechRecognitionServiceUri, SpeechRecognitionAudiostartHandler, SpeechRecognitionSoundstartHandler, SpeechRecognitionSpeechstartHandler, SpeechRecognitionSpeechendHandler, SpeechRecognitionSoundendHandler, SpeechRecognitionAudioendHandler, SpeechRecognitionResultHandler, SpeechRecognitionNomatchHandler, SpeechRecognitionErrorHandler, SpeechRecognitionStartHandler, SpeechRecognitionEndHandler } from './lib/service/speech-recognition.token';
export { resultList, RxSpeechRecognitionService } from './lib/service/rx-speech-recognition.service';
export { SpeechRecognitionService } from './lib/service/speech-recognition.service';
export {} from './lib/speech-recognition.config';
export { SPEECH_RECOGNITION_DEFAULT } from './lib/speech-recognition.provider';
export { SpeechRecognitionModule } from './lib/speech-recognition.module';
